kafka/integration_test.go

```diff
@@ -1084,3 +1088,364 @@ func TestProducerConsumerHeaders(t *testing.T) {
+	if err != nil {
+		t.Fatalf("DeleteTopics() failed: %s", err)
+	}
+
+	validateTopicResult(t, result2, expError)
+}
+
+func validateConfig(t *testing.T, results []ConfigResourceResult, expResults []ConfigResourceResult, checkConfigEntries bool) {
+
+	_, file, line, _ := runtime.Caller(1)
+	caller := fmt.Sprintf("%s:%d", path.Base(file), line)
+
+	if len(results) != len(expResults) {
+		t.Fatalf("%s: Expected %d results, got %d: %v", caller, len(expResults), len(results), results)
+	}
+
+	for i, result := range results {
+		expResult := expResults[i]
+
+		if result.Error.Code() != expResult.Error.Code() {
+			t.Errorf("%s: %v: Expected %v, got %v", caller, result, expResult.Error.Code(), expResult.Error)
+			continue
+		}
+
+		if !checkConfigEntries {
+			continue
+		}
+
+		matchCnt := 0
+		for _, expEntry := range expResult.Config {
+
+			entry, ok := result.Config[expEntry.Name]
+			if !ok {
+				t.Errorf("%s: %v: expected config %s not found in result", caller, result, expEntry.Name)
+				continue
+			}
+
+			if entry.Value != expEntry.Value {
+				t.Errorf("%s: %v: expected config %s to have value \"%s\", not \"%s\"", caller, result, expEntry.Name, expEntry.Value, entry.Value)
+				continue
+			}
+
+			matchCnt++
+		}
+
+		if matchCnt != len(expResult.Config) {
+			t.Errorf("%s: %v: only %d/%d expected configs matched", caller, result, matchCnt, len(expResult.Config))
+		}
+	}
+
+	if t.Failed() {
+		t.Fatalf("%s: ConfigResourceResult validation failed: see previous errors", caller)
+	}
+}
+
+func TestAdminConfig(t *testing.T) {
+	rand.Seed(time.Now().Unix())
+
+	a := createAdminClient(t)
+	defer a.Close()
+
+	// Steps:
+	//  1) Create a topic, providing initial non-default configuration
+	//  2) Read back config to verify
+	//  3) Alter config
+	//  4) Read back config to verify
+	//  5) Delete the topic
+
+	topic := fmt.Sprintf("%s-config-%d", testconf.Topic, rand.Intn(100000))
+
+	// Expected config
+	expResources := []ConfigResourceResult{
+		{
+			Type: ResourceTopic,
+			Name: topic,
+			Config: map[string]ConfigEntryResult{
+				"compression.type": ConfigEntryResult{
+					Name:  "compression.type",
+					Value: "snappy",
+				},
+			},
+		},
+	}
+	// Create topic
+	newTopics := []NewTopic{{
+		Topic:             topic,
+		NumPartitions:     1,
+		ReplicationFactor: 1,
+		Config:            StringMapToConfigEntries(map[string]string{"compression.type": "snappy"}, AlterOperationSet),
+	}}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	topicResult, err := a.CreateTopics(ctx, newTopics, nil)
+	if err != nil {
+		t.Fatalf("Create topic request failed: %v", err)
+	}
+
+	if topicResult[0].Error.Code() != ErrNoError {
+		t.Fatalf("Failed to create topic %s: %s", topic, topicResult[0].Error)
+	}
+
+	// Read back config to validate
+	configResources := []ConfigResource{{Type: ResourceTopic, Name: topic}}
+	describeRes, err := a.DescribeConfigs(ctx, configResources, nil)
```

Comment:
It would be good to test that `DescribeConfigs` on a topic that doesn't exist behaves as expected (I don't think it does).

Reply:
👍

Reply:
Yah, this is tested in librdkafka and it exposed a bug in the broker where it would return the default config for unknown topics, rather than an error.